client/utilities/rfc-index-xml.ts

```diff
@@ -0,0 +1,213 @@
+    arrayNodeName: 'rfc-entry'
+  })
+
+  docListArg.sort = ['number'] // sort by first RFC
+  let offset = 0
+
+  while (!abortController.signal.aborted) {
+    docListArg.offset = offset
+    docListArg.limit = 1000
+
+    const response = await redApi.red.docList(docListArg)
+
+    response.results.forEach((rfcMetadata) => {
+      const rfc = getRFCWithExtraFields(rfcMetadata)
+
+      const [month, year] = DateTime.fromISO(rfcMetadata.published)
+        .toFormat('LLLL yyyy')
+        .split(' ')
+
+      // Based on https://github.com/rfc-editor/rpcwebsite/blob/edf4896c1d97fdd79a78ee6145e3a0c5ffb11fb9/rfc-ed/bin/xmlIndex.pl
+      const rfcForXml = {
+        'rfc-entry': {
+          'doc-id': `RFC${rfc.number}`,
+          title: rfc.title,
+          ...(rfc.authors.length > 0 ?
+            {
+              author: {
+                name: rfc.authors.map(formatAuthor)
+              }
+            }
+          : {}),
+          date: { month, year },
+          ...(rfc.formats ?
+            {
+              format: {
+                'file-format': rfc.formats.map((format) => {
+                  if (format.type === 'TXT') {
+                    return 'ASCII'
+                  }
+                  return format.type
+                })
+              }
+            }
+          : {}),
+          'page-count': rfc.pages,
+          ...(rfc.keywords ?
+            {
+              keywords: {
+                kw: rfc.keywords
+              }
+            }
+          : {}),
+          ...(rfc.obsoletes && rfc.obsoletes.length > 0 ?
+            {
+              obsoletes: {
+                'doc-id': rfc.obsoletes.map(
+                  (obsoletesItem) => `RFC${obsoletesItem.number}`
+                )
+              }
+            }
+          : {}),
+          ...(rfc.obsoleted_by && rfc.obsoleted_by.length > 0 ?
+            {
+              'obsoleted-by': {
+                'doc-id': rfc.obsoleted_by.map(
+                  (obsoletedByItem) => `RFC${obsoletedByItem.number}`
+                )
+              }
+            }
+          : {}),
+          ...(rfc.updated_by && rfc.updated_by.length > 0 ?
+            {
+              'updated-by': {
+                'doc-id': rfc.updated_by.map(
+                  (updatedByItem) => `RFC${updatedByItem.number}`
+                )
+              }
+            }
+          : {}),
+          ...(rfc.abstract ?
+            {
+              abstract:
+                // Certain RFCs have multiple paragraphs such as RFC1849, RFC2169, RFC2178, RFC2660, RFC3318, RFC3867, RFC3873, RFC3875, RFC3886, RFC3890, RFC3894, RFC3903, RFC3918, RFC3919, RFC3927, RFC3932, RFC3933, RFC3938, RFC3945, RFC3949, RFC3965, RFC3974, RFC3987, RFC3995, RFC4005, RFC4010, RFC4025, RFC4026, RFC4034, RFC4035, RFC4040, RFC4041, RFC4042, RFC4043, RFC4053, RFC4061, RFC4078, RFC4080, RFC4082, RFC4086, RFC4088, RFC4090 etc...
+                rfc.abstract.split('\n').map((abstractParagraph) => ({
```

Comment:
Need to de-dup multiple consecutive `\n` here or otherwise prevent possible empty paragraphs? I think there's a good chance paragraphs will have `\n\n` between them unless we adjust on the server side.

Reply:
So it is linebreak separated? I wasn't sure if it was HTML or plaintext w/ `\n`s. Yep I'll make sure to handle `\n\n`. 

Regardless, the tests will verify the amount of paragraph tags against the current XML rendering so it should be ok.

Reply:
I find your abundance of faith in the current XML rendering disturbing. 😁